mapie/conformity_scores/sets/raps.py

```diff
@@ -0,0 +1,427 @@
+                y_pred_proba_raps,
+                quantiles_,
+                include_last_label,
+                lambda_=lambda_,
+                k_star=k_star
+            )
+
+            y_ps = np.greater_equal(
+                    y_pred_proba_raps - y_pred_proba_last, -EPSILON
+            )
+            lambda_star, best_sizes = self._update_size_and_lambda(
+                best_sizes, alpha_np, y_ps, lambda_, lambda_star
+            )
+
+        if len(lambda_star) == 1:
+            lambda_star = lambda_star[0]
+
+        return lambda_star
+
+    def get_conformity_score_quantiles(
+        self,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        estimator: EnsembleClassifier,
+        agg_scores: Optional[str] = "mean",
+        include_last_label: Optional[Union[bool, str]] = True,
+        X_raps: Optional[NDArray] = None,
+        y_raps_no_enc: Optional[NDArray] = None,
+        y_pred_proba_raps: Optional[NDArray] = None,
+        position_raps: Optional[NDArray] = None,
+        **kwargs
+    ) -> NDArray:
+        """
+        Get the quantiles of the conformity scores for each uncertainty level.
+
+        Parameters:
+        -----------
+        conformity_scores: NDArray of shape (n_samples,)
+            Conformity scores for each sample.
+
+        alpha_np: NDArray of shape (n_alpha,)
+            NDArray of floats between 0 and 1, representing the uncertainty
+            of the confidence interval.
+
+        estimator: EnsembleClassifier
+            Estimator that is fitted to predict y from X.
+
+        agg_scores: Optional[str]
+            Method to aggregate the scores from the base estimators.
+            If "mean", the scores are averaged. If "crossval", the scores are
+            obtained from cross-validation.
+
+            By default, ``"mean"``.
+
+        include_last_label: Optional[Union[bool, str]]
+            Whether or not to include last label in prediction sets.
+            Choose among ``False``, ``True``  or ``"randomized"``.
+
+            By default, ``True``.
+
+        X_raps: NDArray of shape (n_samples, n_features)
+            Observed feature values for the RAPS method (split data).
+
+            By default, "None" but must be set to work.
+
+        y_raps_no_enc: NDArray of shape (n_samples,)
+            Observed labels for the RAPS method (split data).
+
+            By default, "None" but must be set to work.
+
+        y_pred_proba_raps: NDArray of shape (n_samples, n_classes)
+            Predicted probabilities for the RAPS method (split data).
+
+            By default, "None" but must be set to work.
+
+        position_raps: NDArray of shape (n_samples,)
+            Position of the points in the split set for the RAPS method
+            (split data). These positions are returned by the function
+            ``get_true_label_position``.
+
+            By default, "None" but must be set to work.
+
```

Comment:
Don't forget to add `**kwargs`. Note that there seems like there are multiple times where it might have been forgotten.

Reply:
It's a question whether we want to add it everywhere or not. @vincentblot28 @thibaultcordier ?

Reply:
I use **kwargs to allow increasing the size of input parameters in child classes without the parent class writing a warning. In practice, the user does not need to know that **kwargs if using the target class.

Reply:
It's just a developer trick, which does nothing for the end user except confusion. I prefer not to put that.